wallet/wallet.go

```diff
@@ -2359,6 +2359,149 @@ func (w *Wallet) GetTransactions(startBlock, endBlock *BlockIdentifier,
 	return &res, err
 }
 
+// GetTransaction returns data if any transaction given its id. A mined
+// transaction is returned in a Block structure which records properties about
+// the block. A bool is also returned to denote if the transaction did exist
+// in the database or not.
+func (w *Wallet) GetTransaction(txHash *chainhash.Hash) (*GetTransactionsResult,
```

Comment:
Hm. Looking at the way you use this method from `lnd-strike` and also the fact that you always return either one mined or one unmined txn, I think this is not the most appropriate return type. We probably need a new type for this, maybe something like
```go
type GetTransactionResult struct {
	MinedTransactions   *Block
	UnminedTransactions *TransactionSummary
}
```
And also we can document that at least one of the fields will be null. Possibly both if we can't find the txn.

Reply:
Good point! I was a bit reluctant to do that as I thought this would involve changing the protobuf and we can simply declare the type.